src/libs/ck-libs/allGather/allGather.ci

```diff
@@ -0,0 +1,59 @@
+module allGather
+{
+  include "vector";
+
+  message allGatherMsg;
+
+  group AllGather
+  {
+    entry AllGather(int size, int type, int seed);
+    entry[reductiontarget] void startGather();
+    entry void recvRing(int sender, CkNcpyBuffer data);
+    entry[nokeep] void local_buff_done(CkDataMsg * m);
+    entry void Hypercube()
+    {
+      for (iter = 0; iter < numHypercubeIter; iter++)
+      {
+        serial
+        {
+          if (HypercubeRecursiveDoubling)
+            HypercubeToSend = (n + CkMyPe() - (int)pow(2, iter)) % n;
+          else
+            HypercubeToSend = CkMyPe() ^ ((int)pow(2, iter));
+        }
+        serial
+        {
+          if (HypercubeRecursiveDoubling && iter == numHypercubeIter - 1)
+          {
+            int size = n - (int)pow(2, iter);
+            if(size > hyperCubeIndxCopy.size()){
```

Comment:
I don't think you need hyperCubeIndxCopy and hyperCubeStoreCopy at all actually, can't you just use hyperCubeIndx and hyperCubeStore in the call to recvHypercube()?